gym/utils/env_checker.py

```diff
@@ -67,31 +68,65 @@ def check_reset_seed(env: gym.Env):
-    if "seed" in signature.parameters or "kwargs" in signature.parameters:
+    if "seed" in signature.parameters or (
+        "kwargs" in signature.parameters
+        and signature.parameters["kwargs"].kind is inspect.Parameter.VAR_KEYWORD
+    ):
         try:
             obs_1 = env.reset(seed=123)
-            assert obs_1 in env.observation_space
-            obs_2 = env.reset(seed=123)
-            assert obs_2 in env.observation_space
-            assert data_equivalence(obs_1, obs_2)
-            seed_123_rng = deepcopy(env.unwrapped.np_random)
+            assert (
+                obs_1 in env.observation_space
+            ), "The observation returned by `env.reset(seed=123)` is not within the observation space"
+            assert (
+                # pyright: ignore [reportPrivateUsage]
+                env.unwrapped._np_random
+                is not None
+            ), "Expects the random number generator to have been generated given a seed was passed to reset. Mostly likely the environment reset function does not call `super().reset(seed=seed)`."
+            seed_123_rng = deepcopy(
+                env.unwrapped._np_random  # pyright: ignore [reportPrivateUsage]
+            )
 
-            # Note: for some environment, they may initialise at the same state, therefore we cannot check the obs_1 != obs_3
-            obs_4 = env.reset(seed=None)
-            assert obs_4 in env.observation_space
+            obs_2 = env.reset(seed=123)
+            assert (
+                obs_2 in env.observation_space
+            ), "The observation returned by `env.reset(seed=123)` is not within the observation space"
+            if env.spec is not None and env.spec.nondeterministic is False:
+                assert data_equivalence(
+                    obs_1, obs_2
+                ), "`env.reset(seed=123)` is not deterministic as the observations are not equivalent"
+            assert (
+                env.unwrapped._np_random.bit_generator.state  # pyright: ignore [reportPrivateUsage]
+                == seed_123_rng.bit_generator.state
+            ), (
+                "Mostly likely the environment reset function does not call `super().reset(seed=seed)` "
+                "as the random generates are not same when the same seeds are passed to `env.reset`."
+            )
 
+            obs_3 = env.reset(seed=456)
+            assert (
+                obs_3 in env.observation_space
+            ), "The observation returned by `env.reset(seed=456)` is not within the observation space"
             assert (
-                env.unwrapped.np_random.bit_generator.state
+                # pyright: ignore [reportPrivateUsage]
+                env.unwrapped._np_random
+                is not None
+            ), "Expects the random number generator to have been generated given a seed was passed to reset. Mostly likely the environment reset function does not call `super().reset(seed=seed)`."
```

Comment:
Not sure whether we need it because we have repeatedly checked in the try block that `env.unwrapped._np_random` is not None

Reply:
Removed